scheduler/src/cook/mesos.clj

```diff
@@ -233,16 +235,17 @@
                                                                               :trigger-chan rebalancer-trigger-chan
                                                                               :view-incubating-offers view-incubating-offers})
                                     (when (seq optimizer-config)
-                                      (cook.mesos.optimizer/start-optimizer-cycles! (fn get-queue []
-                                                                                      ;; TODO Use filter of queue that scheduler uses to filter to considerable.
-                                                                                      ;;      Specifically, think about filtering to jobs that are waiting and 
-                                                                                      ;;      think about how to handle quota 
-                                                                                      @pool-name->pending-jobs-atom)
-                                                                                    (fn get-running []
-                                                                                      (cook.mesos.util/get-running-task-ents (d/db mesos-datomic-conn)))
-                                                                                    view-incubating-offers
-                                                                                    optimizer-config
-                                                                                    optimizer-trigger-chan))
+                                      (optimizer/start-optimizer-cycles!
+                                        optimizer-config
+                                        optimizer-trigger-chan
+                                        (fn get-pool-names []
+                                          (keys @pool-name->pending-jobs-atom))
+                                        (fn pool-name->queue [pool-name]
+                                          (get @pool-name->pending-jobs-atom pool-name))
+                                        (fn pool-name->running [pool-name]
+                                          (->> (util/get-running-task-ents (d/db mesos-datomic-conn))
+                                               (filter #(= pool-name (-> % :job/_instance util/job->pool-name)))))
```

Comment:
Would it make sense to move this filtering logic into the query? E.g., we could add an optional `:pool-name` argument to the `get-running-task-ents` function, when then gets optionally added to the query. 

If we decide to update `get-running-task-ents`, that should probably go into its own PR, and we could rebase this on top.

Reply:
Why did we favor filtering outside the query?

Reply:
I spoke with @pschorf and @dposada and they are okay with moving the pool inside the query. I created #1002

Reply:
Works for me